src/OrleansAWSUtils/Membership/DynamoDBMembershipTable.cs

```diff
@@ -0,0 +1,350 @@
+﻿using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.Model;
+using Orleans.Runtime.Configuration;
+using OrleansAWSUtils.Storage;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.Runtime.MembershipService
+{
+    internal class DynamoDBMembershipTable : IMembershipTable
+    {
+        //DynamoDB does not support the extended Membership Protocol and will always return the same table version information
+        private readonly TableVersion _tableVersion = new TableVersion(0, "0");
+
+        private const string TABLE_NAME_DEFAULT_VALUE = "OrleansSiloInstances";
+        private const string CURRENT_ETAG_ALIAS = ":currentETag";
+
+        private Logger logger;
+        private DynamoDBStorage storage;
+        private string deploymentId;
+
+        public Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, Logger log)
+        {
+            logger = log;
+            deploymentId = config.DeploymentId;
+            storage = new DynamoDBStorage(config.DataConnectionString, log);
+            logger.Info(ErrorCode.MembershipBase, "Initializing AWS DynamoDB Membership Table");
+            return storage.InitializeTable(TABLE_NAME_DEFAULT_VALUE,
+                new List<KeySchemaElement>
+                {
+                    new KeySchemaElement { AttributeName = SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME, KeyType = KeyType.HASH },
+                    new KeySchemaElement { AttributeName = SiloInstanceRecord.SILO_IDENTITY_PROPERTY_NAME, KeyType = KeyType.RANGE }
+                },
+                new List<AttributeDefinition>
+                {
+                    new AttributeDefinition { AttributeName = SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME, AttributeType = ScalarAttributeType.S },
+                    new AttributeDefinition { AttributeName = SiloInstanceRecord.SILO_IDENTITY_PROPERTY_NAME, AttributeType = ScalarAttributeType.S }
+                });
+        }
+
+        public async Task DeleteMembershipTableEntries(string deploymentId)
+        {
+            try
+            {
+                var keys = new Dictionary<string, AttributeValue> { { $":{SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME}", new AttributeValue(deploymentId) } };
+                var records = await storage.QueryAsync(TABLE_NAME_DEFAULT_VALUE, keys, $"{SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME} = :{SiloInstanceRecord.DEPLOYMENT_ID_PROPERTY_NAME}", item => new SiloInstanceRecord(item));
+
+                var toDelete = new List<Dictionary<string, AttributeValue>>();
+                foreach (var record in records)
+                {
+                    toDelete.Add(record.GetKeys());
+                }
+
+                if (records.Count <= 25)
```

Comment:
I don't think you need a special case here, the general loop below is good. Also, 25? at least lets use constant.